service/controller/v21/resource/cloudformation/current.go

```diff
@@ -217,22 +216,53 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 			return StackState{}, microerror.Mask(err)
 		}
 
-		workerCount, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerCountKey)
+		workerASGName, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerASGKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
-		workerImageID, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerImageIDKey)
+		workerCloudConfigVersion, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerCloudConfigVersionKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
-		workerInstanceType, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerInstanceTypeKey)
+		workerImageID, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerImageIDKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
-		workerCloudConfigVersion, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerCloudConfigVersionKey)
+		workerInstanceType, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerInstanceTypeKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
+		var workerDesired, workerMin, workerMax int
+		{
+			r.logger.LogCtx(ctx, "level", "debug", "message", "finding out worker asg properties")
+
+			i := &autoscaling.DescribeAutoScalingGroupsInput{
+				AutoScalingGroupNames: []*string{
+					&workerASGName,
+				},
+			}
+			o, err := ctlCtx.AWSClient.AutoScaling.DescribeAutoScalingGroups(i)
+			if err != nil {
+				return StackState{}, microerror.Mask(err)
+			}
+			if len(o.AutoScalingGroups) == 0 {
+				return StackState{}, microerror.Maskf(notExistsError, "asg for name %s", workerASGName)
```

Comment:
We usually have `notExistsError`. But isn't it isn't handled anywhere I'd replace them with `ececutionFailedError` to make that notion clear.

Reply:
Ack